includes/learnopengl/model.h

```diff
@@ -173,6 +212,41 @@ class Model
     vector<Texture> loadMaterialTextures(aiMaterial *mat, aiTextureType type, string typeName)
     {
         vector<Texture> textures;
+        
+        // First check if there isn't a texture, if so create an empty image
+        if (mat->GetTextureCount(type) == 0)
+        {
+            bool skip = false;
+            for (unsigned int j = 0; j < textures_loaded.size(); j++)
+            {
+                if (std::strcmp(textures_loaded[j].path.data(), "NO_TEXTURE") == 0)
+                {
+                    textures.push_back(textures_loaded[j]);
+                    skip = true; // A texture with the same filepath has already been loaded, continue to next one. (optimization)
+                    break;
+                }
+            }
+            if (!skip)
+            {   // If texture hasn't been loaded already, load it
+                Texture texture;
+    
+                if (typeName == "texture_diffuse")
+                    texture.id = noDiffuse; // Use the default diffuse map
+                else if (typeName == "texture_specular")
```

Comment:
You can actually remove this part. Less logic required possibly.

Reply:
Good catch